packages/next/build/swc/options.js

```diff
@@ -0,0 +1,126 @@
+const nextDistPath =
+  /(next[\\/]dist[\\/]shared[\\/]lib)|(next[\\/]dist[\\/]client)|(next[\\/]dist[\\/]pages)/
+
+function getBaseSWCOptions({
+  filename,
+  development,
+  hasReactRefresh,
+  globalWindow,
+}) {
+  const isTSFile = filename.endsWith('.ts')
+  const isTypeScript = isTSFile || filename.endsWith('.tsx')
+
+  return {
+    jsc: {
+      parser: {
+        syntax: isTypeScript ? 'typescript' : 'ecmascript',
+        dynamicImport: true,
+        // Exclude regular TypeScript files from React transformation to prevent e.g. generic parameters and angle-bracket type assertion from being interpreted as JSX tags.
+        [isTypeScript ? 'tsx' : 'jsx']: isTSFile ? false : true,
+      },
+
+      transform: {
+        react: {
+          runtime: 'automatic',
+          pragma: 'React.createElement',
+          pragmaFrag: 'React.Fragment',
+          throwIfNamespace: true,
+          development: development,
+          useBuiltins: true,
+          refresh: hasReactRefresh,
+        },
+        optimizer: {
+          simplify: false,
+          globals: {
+            typeofs: {
+              window: globalWindow ? 'object' : 'undefined',
+            },
+          },
+        },
+        regenerator: {
+          importPath: require.resolve('regenerator-runtime'),
+        },
+      },
+    },
+  }
+}
+
+export function getJestSWCOptions({ filename, esm }) {
+  let baseOptions = getBaseSWCOptions({
+    filename,
+    development: false,
```

Comment:
@timneutkens you mentioned we should be using the browser config, but there were a few things that seemed like they should be different. This is the first one I noticed based on the options we were passing before in `jest.config.js`

Reply:
That's fine 👍